### ISSUE TITLE
Incorrect result for std::sqrt and other cmath functions on long double types.

### ISSUE THREAD

**Issue by jprendes** [emscripten-core/emscripten#10516]
The following code (in `main.cpp`) produces wrong answers when compiled with emscripten.
```c++
#include <cmath>
#include <string>
#include <iostream>

int main(int argc, char **argv) {
    long double x = std::stold(argc > 1 ? argv[1] : "16.0");
    std::cout << "x = " << x << std::endl;
    std::cout << "sqrt(x) = " << std::sqrt(x) << std::endl;
    std::cout << "ceil(x) = " << std::ceil(x) << std::endl;
    std::cout << "floor(x) = " << std::floor(x) << std::endl;
    std::cout << "atan(x) = " << std::atan(x) << std::endl;
    std::cout << "atan2(x,1) = " << std::atan2(x,1) << std::endl;
    std::cout << "asin(x) = " << std::asin(x) << std::endl;
    std::cout << "acos(x) = " << std::acos(x) << std::endl;
    return 0;
}
```

I am compiling using the following flags:
```bash
em++ -O2 -o./build/index.js main.cpp
```

Running on node `node build/index.js 0.6` I obtain the following output
```
x = 0.6
sqrt(x) = 0
ceil(x) = 0
floor(x) = 0
atan(x) = -5.34172e-25
atan2(x,1) = -5.34172e-25
asin(x) = -inf
acos(x) = 1.27351
```
floor always produces 0, for any input.
I obtain the same output running on Chrome, Firefox, or Node.

The output I would expect is
```
x = 0.6
sqrt(x) = 0.774597
ceil(x) = 1
floor(x) = 0
atan(x) = 0.54042
atan2(x,1) = 0.54042
asin(x) = 0.643501
acos(x) = 0.927295
```

Replacing `long double` with `double` produces the right output.
Compiling to native with clang produces the right output.

Here are the bitecode, html, js and wasm files: [index.zip](https://github.com/emscripten-core/emscripten/files/4232186/index.zip)

The output of `emcc -v` is
```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.39.8
clang version 11.0.0 (/b/s/w/ir/cache/git/chromium.googlesource.com-external-github.com-llvm-llvm--project 9dc84e9b02d1e402503906099d42fbae4da7d8d9)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /local-ssd/Projects/EmscriptenLongDouble/cache/emsdk/upstream/bin
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6.3.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.3.0
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.3.0
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Candidate multilib: x32;@mx32
Selected multilib: .;@m64
shared:INFO: (Emscripten: Running sanity checks)
```

**Comment by kripken** [comment]
Thanks @jprendes ! I see the same.

This looks like a bug with `long double` since it works without the `long`. So something in the emulation of float128 is not working it seems.

Is it too late to make `long double` just be `double`? :wink:

**Comment by jprendes** [comment]
Making `long double` == `double` would be fine in my case.
It would at least be usable ;-)

From what I've seen, in the current situation the generated code wasm file (I guess from musl?) ends up calling to JS's Math.sqrt with 3 arguments (I think those are a memory location, and two 64 bits numbers that make up the 128 bit float). I guess fixing that emulation is hard.

**Comment by kripken** [comment]
Turns out this was due to some missing files in our system libraries. Fix in #10565
